### PR TITLE
MM-52944: Fix profile popover view issue on RHS when searching channel member

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
This PR is to fix an issue where clicking an avatar shows profile popover only a few seconds. This behavior happens only when searching for a user

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/23862
Jira https://mattermost.atlassian.net/browse/MM-52944

#### Screenshots

**before**

https://github.com/mattermost/mattermost/assets/37421564/a1bdf88f-cd57-4dd8-83cd-7c5760761671 

**after**

https://github.com/mattermost/mattermost/assets/37421564/2118e1c0-30e4-4a88-832b-5313056b38bd

#### Release Note
```release-note
Fixes a UI issue related to profile popover on channel member search in right hand side panel
```

### CODE REVIEW

File: webapp/channels/src/components/at_sum_members_mention/notification_from_members_modal.tsx [hunk 1]
```diff
@@ -91,13 +91,13 @@ function NotificationFromMembersModal(props: Props) {
         };
     });
 
-    const openDirectMessage = async (user: UserProfile) => {
+    const openDirectMessage = useCallback(async (user: UserProfile) => {
```
Comment (sinansonmez) [inline webapp/channels/src/components/at_sum_members_mention/notification_from_members_modal.tsx:94]: This was actually not needed for this issue but I thought would be a nice improvement. I could not find where this component is being used so I could not test. Please test it on QA test

**Comment by sinansonmez** [comment]
kindly pinging fro review @mgdelacroix @crspeller

File: webapp/channels/src/components/channel_members_rhs/channel_members_rhs.tsx [hunk 2]
```diff
@@ -150,7 +150,9 @@ export default function ChannelMembersRHS({
 
             listcp.push({type: ListItemType.Member, data: member});
         }
-        setList(listcp);
+        if (JSON.stringify(list) !== JSON.stringify(listcp)) {
+            setList(listcp);
+        }
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/channel_members_rhs/channel_members_rhs.tsx:155]: if we add `list` as dependency do we still need to do this?

File: webapp/channels/src/components/channel_members_rhs/member_list.tsx [hunk 3]
```diff
@@ -19,21 +19,19 @@ export interface Props {
...
-    actions: {
-        openDirectMessage: (user: UserProfile) => void;
-        loadMore: () => void;
-    };
+    openDirectMessage: (user: UserProfile) => void;
+    loadMore: () => void;
 }
 
-const MemberList = ({
+const MemberList = React.memo(({
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/channel_members_rhs/member_list.tsx:26]: nit : instead of here lets just do
```
export default Memo(MemberList);
```

**Comment by calebroseland** [comment]
> I wonder why the popover disappears on re-render in the first place?

I think when a profile popover opens, `getMembershipForEntities` is fired which causes `channelMembers` to change here.

**Comment by sinansonmez** [comment]
> > I wonder why the popover disappears on re-render in the first place?
> 
> I think when a profile popover opens, `getMembershipForEntities` is fired which causes `channelMembers` to change here.

Exactly that is the behaviour I was seeing

File: webapp/channels/src/components/channel_members_rhs/channel_members_rhs.tsx (hunk 2 above)
Comment (sinansonmez) [reply webapp/channels/src/components/channel_members_rhs/channel_members_rhs.tsx:155]: Unfortunaly not working. adding list as dependency breaks the page when members button is clicked in RHS.

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/channel_members_rhs/channel_members_rhs.tsx:155]: I see thank you for trying it out.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @sinansonmez - this is fixed/working as expected 👍 
cc// @M-ZubairAhmed

**Pull request merged**